### PR TITLE
Register ClusterDicts

### PR DESCRIPTION
**Pull request by amitmurthy**
Implements `GlobalDicts` which sets the same key-value pair on all participating workers and `DistributedDicts` which distributed the KV pairs over participating workers (like memcached).

Will merge in a day in the absence of any objections

### CODE REVIEW

File: ClusterDicts/versions/0.1.0/requires [hunk 1]
```diff
@@ -0,0 +1,3 @@
+julia 0.4-
```
Comment (tkelman) [inline ClusterDicts/versions/0.1.0/requires]: do you need to support prereleases?

**Comment by tkelman** [comment]
Turn Travis on?

File: ClusterDicts/versions/0.1.0/requires (hunk 1 above)
Comment (amitmurthy) [inline ClusterDicts/versions/0.1.0/requires]: copy-paste error from an existing package. Thanks.

**Comment by amitmurthy** [comment]
Yes, will do.

**Pull request merged**